server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1693,23 +1693,21 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
      * @throws IllegalArgumentException if there was an error parsing the value from the json
      * @throws IOException if there was any other IO error
      */
-    private static Number value(XContentParser parser, NumberType numberType, Number nullValue, boolean coerce)
-        throws IllegalArgumentException, IOException {
-
+    public Number value(XContentParser parser) throws IllegalArgumentException, IOException {
```

Comment:
This'll need javadoc if we want to keep it. It's sort of leaky, but I've done worse. It's for a good cause.

Reply:
Maybe @romseygeek should have a look here too.

Reply:
Yep fine by me.